packages/neuron-ui/src/components/LockInfoDialog/index.tsx

```diff
@@ -31,49 +29,27 @@ const ShortAddr = ({ lockScript, isMainnet }: { lockScript: CKBComponents.Script
     codeHashOrCodeHashIndex: lock.CodeHashIndex,
   })
 
-  return (
-    <>
-      <h2 title={t('transaction.deprecated-address-format')} className={styles.title}>
-        {t('transaction.deprecated-address-format')}
-      </h2>
-      <div style={{ marginBottom: 10 }}>
-        <CopyZone content={shortAddr} name={t('history.copy-address')}>
-          <span>{shortAddr}</span>
-        </CopyZone>
-      </div>
-    </>
-  )
+  return shortAddr
 }
 
 const LockInfoDialog = ({ lockInfo, isMainnet, onDismiss }: LockInfoDialogProps) => {
   const [t] = useTranslation()
   const [copied, setCopied] = useState(false)
-  const dialogRef = useRef<HTMLDialogElement | null>(null)
-  useDialog({ show: !!lockInfo, dialogRef, onClose: onDismiss })
 
   const timer = useRef<ReturnType<typeof setTimeout>>()
 
-  const onDialogClicked = useCallback(
-    (e: any) => {
-      if (e.target.tagName === 'DIALOG') {
-        onDismiss()
-      }
-    },
-    [onDismiss]
-  )
-  if (!lockInfo) {
-    return null
-  }
-
-  const rawLock = `{
+  const rawLock = lockInfo
+    ? `{
   "code_hash": "${lockInfo.codeHash}",
   "hash_type": "${lockInfo.hashType}",
   "args": "${lockInfo.args}"
 }`
+    : ''
+  const shortAddr = getShortAddr({ lockScript: lockInfo, isMainnet }) ?? ''
```

Comment:
I think we can use `useMemo` to cache `shortAddr`.

Reply:
resolved